File: tests/test_dart/lib/services/order_service.dart
import 'package:intl/intl.dart';
import '../models/order.dart';
import '../models/user.dart';

class OrderService {
  bool process(Order order, User? user) {
    print('Processing order ${order.id}');
    if (user != null) {
      print('For user: ${user.name}');
    }
    print('Total: ${order.formatTotal()}');
    return true;
  }

  bool cancel(int orderId) {
    final now = DateTime.now();
    final formatter = DateFormat('yyyy-MM-dd HH:mm:ss');
    print('Canceling order $orderId at ${formatter.format(now)}');
    return true;
  }

  List<Order> getOrdersByUser(User user) {
    return [];
  }
}

File: tests/test_elixir/lib/my_app/services/order_service.ex
defmodule MyApp.Services.OrderService do
  alias MyApp.Models.Order
  alias MyApp.Models.User
  require Logger

  def process(%Order{} = order, %User{} = user) do
    IO.puts("Processing order #{order.id} for user #{user.name}")
    IO.puts("Total: #{Order.format_total(order)}")
    true
  end

  def cancel(order_id) do
    now = DateTime.utc_now() |> DateTime.to_string()
    IO.puts("Canceling order #{order_id} at #{now}")
    true
  end

  def get_orders_by_user(%User{} = user) do
    []
  end
end
